packages/mds-db/events.ts

```diff
@@ -304,6 +304,92 @@ export async function readEventsWithTelemetry({
   }))
 }
 
+// TODO: remove
+// heinous copypasta specifically to provide the VIN with every event, used ONLY by Native.
+// this is to be excised when we dump Native in the garbage in the 1.0 timeframe.
+export async function readEventsWithTelemetryAndVehicleId({
+  device_id,
+  provider_id,
+  start_time,
+  end_time,
+  order_by = 'id',
+  last_id = 0,
+  limit = 1000
+}: Partial<{
+  device_id: UUID
+  provider_id: UUID
+  start_time: Timestamp
+  end_time: Timestamp
+  order_by: string
+  last_id: number
+  limit: number
+}>): Promise<Recorded<VehicleEvent & Pick<Device, 'vehicle_id'>>[]> {
+  const client = await getReadOnlyClient()
+  const vals = new SqlVals()
+  const exec = SqlExecuter(client)
+
+  const conditions: string[] = last_id ? [`id > ${vals.add(last_id)}`] : []
+
+  if (provider_id) {
+    if (!isUUID(provider_id)) {
+      throw new Error(`invalid provider_id ${provider_id}`)
+    } else {
+      conditions.push(`provider_id = ${vals.add(provider_id)}`)
+    }
+  }
+
+  if (device_id) {
+    if (!isUUID(device_id)) {
+      throw new Error(`invalid device_id ${device_id}`)
+    } else {
+      conditions.push(`device_id = ${vals.add(device_id)}`)
+    }
+  }
+
+  if (start_time !== undefined) {
+    if (!isTimestamp(start_time)) {
+      throw new Error(`invalid start_time ${start_time}`)
+    } else {
+      conditions.push(`timestamp >= ${vals.add(start_time)}`)
+    }
+  }
+
+  if (end_time !== undefined) {
+    if (!isTimestamp(end_time)) {
+      throw new Error(`invalid end_time ${end_time}`)
+    } else {
+      conditions.push(`timestamp <= ${vals.add(end_time)}`)
+    }
+  }
+
+  // we can only select based on event criteria
+  const where = conditions.length ? ` WHERE ${conditions.join(' AND ')}` : ''
+
+  const { rows } = await exec(
+    `SELECT E.*, D.vehicle_id, T.lat, T.lng, T.speed, T.heading, T.accuracy, T.altitude, T.charge, T.timestamp AS telemetry_timestamp 
+      FROM (SELECT * FROM ${schema.TABLE.events}${where} LIMIT ${vals.add(limit)}) AS E 
```

Comment:
The events _need_ to have this inner `ORDER BY` in order for the cursor to sequence properly and get the proper `LIMIT` number of rows based on `"id"`. After joining, it's possible that PG shuffles the deck again; hence the outer ORDER BY, IIRC.

Reply:
Now that the method has been copy/pasta'd, I believe `audit` is the only remaining caller of the original and I am pretty sure it doesn't need any of the last_id/limit behavior so it could potentially be simplified.